<launch>
  <!-- Point cloud topic to subscribe to -->
  <arg name="cloud_topic" default="/sensor_data/raw_data/point_cloud"/>
  <!-- Image topic to subscribe to -->
  <arg name="image_topic" default="/sensor_data/raw_data/camera"/>
  <!-- Cone topic to publish to -->
  <arg name="cone_topic" default="/perception/cone_image_map"/>
  <!-- Use simulated sensors -->
  <arg name="simulating" default="false"/>
  <!-- Use bag file -->
  <arg name="bag_file" default="false"/>
  <!-- Display opencv window -->
  <arg name="display" default="false"/>
  <!-- Include image detection launch file -->
  <include file="$(find perception_meta)/launch/camera_perception.launch">
    <arg name="simulating" value="$(arg simulating)"/>
    <arg name="cloud_topic" value="$(arg cloud_topic)"/>
    <arg name="image_topic" value="$(arg image_topic)"/>
    <arg name="bag_file" value="$(arg bag_file)"/>
  </include>
  <group ns="perception">
    <group ns="point_cloud">
      <!-- Launch thresholding node -->
      <node pkg="point_cloud_thresholder" type="point_cloud_thresholder" name="point_cloud_thresholder" output="screen" required="true">
        <param name="cloud_topic" value="$(arg cloud_topic)"/>
        <param name="cone_topic" type="str" value="$(arg cone_topic)"/>
        <param name="simulating" type="bool" value="$(arg simulating)"/>
        <param name="display" type="bool" value="$(arg display)"/>
      </node>
    </group>
  </group>
</launch>
